test/DynamoCoreTests/Nodes/IfTest.cs

```diff
@@ -76,5 +80,102 @@ public void TestScopeIfForPreview()
             AssertPreviewValue("9fe8e82f-760d-43a6-90b2-5f9c252139d7", 42);
             AssertPreviewValue("23a03082-5807-4e44-9a3d-2d1eec4a914c", 42);
         }
+
+        [Test]
+        [Category("SmokeTest")]
+        public void TestIfNodeForPreview()
+        {
+            string testFilePath = Path.Combine(testFolder, "testNewIf.dyn");
+            OpenModel(testFilePath);
+            BoolSelector boolSelectorNode = (BoolSelector) this.CurrentDynamoModel.CurrentWorkspace.NodeFromWorkspace("886a464b-9b2b-4e66-a033-c18d0753c2cf");
+
+            boolSelectorNode.Value = true;
+            AssertPreviewValue("886a464b-9b2b-4e66-a033-c18d0753c2cf", true);
+            AssertPreviewValue("f945529e-62e3-4c7a-b07a-0b18d7449f9b", new object[] { "a1", "b1", "c1" });
+            // The output preview for partially connected "IF" node should return the function itself irrespective of test value.
+            AssertPreviewValue("e1630959-8338-4b1a-9832-248246aef8a0", "Function");
+
+            boolSelectorNode.Value = false;
+            AssertPreviewValue("886a464b-9b2b-4e66-a033-c18d0753c2cf", false);
+            AssertPreviewValue("f945529e-62e3-4c7a-b07a-0b18d7449f9b", new object[] { new object[] { "a2" }, "b2" });
+            // The output preview for partially connected "IF" node should return the function itself irrespective of test value.
+            AssertPreviewValue("e1630959-8338-4b1a-9832-248246aef8a0", "Function");
```

Comment:
This is fine but more importantly, we need a test where the `If` node is used as a partially applied function input to another node and ensure that the downstream node works as expected.

Reply:
Modified the test accordingly.

Reply:
I'm looking for an example like this where the partially applied `If` node acts like a `function`, which when applied to a higher order node accepting a `function` returns a valid result. In this example, the `If` "function" is called twice, once for `true`, and secondly for `false` in the other list:
![image](https://user-images.githubusercontent.com/5710686/120736530-67d67080-c4ba-11eb-9888-201715299d5f.png)

Reply:
Got it. Updated the test with the above example.